src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1147,6 +1205,17 @@ private AuthenticationResult acquireTokenAfterValidation(CallbackHandler callbac
             AuthenticationResult result = null;
             request.setVersion(getVersionName());
             request.setBrokerAccountName(request.getLoginHint());
+            
+            //check if App gives permission to AccountManager
+            try
```

Comment:
Am I missing anything? when brokerProxy checks for canSwitchToBroker, we do the verification for permission, then if permission is already missed, this flow won't be hit even.